cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -2317,4 +2317,342 @@
+  </property>
+
+  <property>
+    <name>spark.retry.policy.max.time.secs</name>
+    <value>600</value>
+    <description>
+      The max elapsed time in seconds before retries are aborted.
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.max.retries</name>
+    <value>1000</value>
+    <description>
+      The max number of retries to attempt before aborting.
+      By default, the max time will be hit before this will be hit
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.delay.base.ms</name>
+    <value>1000</value>
+    <description>
+      The base delay between retries in milliseconds. Used by the 'fixed.delay' and 'exponential.backoff' policy types
+    </description>
+  </property>
+
+  <property>
+    <name>spark.retry.policy.delay.max.ms</name>
+    <value>30000</value>
+    <description>
+      The maximum delay between retries in milliseconds. Only used if the policy type is 'exponential.backoff'
+    </description>
+  </property>
+
+  <!-- Custom Action configuration -->
+
+  <property>
+    <name>custom.retry.policy.type</name>
+    <value>exponential.backoff</value>
+    <description>
+      The type of retry policy. Must be 'none', 'fixed.delay', or 'exponential.backoff'.
+      The retry policy is used when CDAP attempted to perform an operation, but that operation failed in a retryable
+      way. The most common type of failure is if a dependent service was unavailable. For example, if a transaction
+      could not be started because the transaction service is not available. Any CDAP method that throws a
+      RetryableException will use this policy if it encounters a RetryableException
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.max.time.secs</name>
+    <value>600</value>
+    <description>
+      The max elapsed time in seconds before retries are aborted.
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.max.retries</name>
+    <value>1000</value>
+    <description>
+      The max number of retries to attempt before aborting.
+      By default, the max time will be hit before this will be hit
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.delay.base.ms</name>
+    <value>1000</value>
+    <description>
+      The base delay between retries in milliseconds. Used by the 'fixed.delay' and 'exponential.backoff' policy types
+    </description>
+  </property>
+
+  <property>
+    <name>custom.retry.policy.delay.max.ms</name>
+    <value>30000</value>
+    <description>
+      The maximum delay between retries in milliseconds. Only used if the policy type is 'exponential.backoff'
+    </description>
+  </property>
+
+  <!-- Workflow configuration -->
+
+  <property>
+    <name>workflow.retry.policy.type</name>
+    <value>exponential.backoff</value>
+    <description>
+      The type of retry policy. Must be 'none', 'fixed.delay', or 'exponential.backoff'.
```

Comment:
this will be become difficult to maintain if we add another retry policy. Is there a way we could maintain the valid values in one place and all other refer to that? @awholegunch what do you think?